src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java

```diff
@@ -0,0 +1,22 @@
+package io.apimatic.coreinterfaces.type.functional;
+
+import java.io.IOException;
+import io.apimatic.coreinterfaces.http.Context;
+
+
+/**
+* Functional Interface to apply the context initialization function.
+*
+* @param <ResponseType> The type of the response in which context initialised.
+*/
+@FunctionalInterface
+public interface ContextInitializer<ResponseType> {
+   /**
+    * 
+    * @param context the context of the api call
+    * @param response the response from the server
+    * @return the Response
+    * @throws IOException which occurs in case of I/O error
+    */
+   ResponseType apply(Context context, ResponseType response) throws IOException;
```

Comment:
Is it expected to throw IOException while applying the function?

Reply:
Removed the throws clause